ibm/service/power/resource_ibm_pi_instance.go

```diff
@@ -1436,8 +1436,18 @@ func createSAPInstance(d *schema.ResourceData, sapClient *instance.IBMPISAPInsta
 		userData := u.(string)
 		body.UserData = encodeBase64(userData)
 	}
-	if sys, ok := d.GetOk(Arg_SysType); ok {
-		body.SysType = sys.(string)
+
+	if deploymentTarget, ok := d.GetOk(Arg_DeploymentTarget); ok {
+		if sys, ok := d.GetOk(Arg_SysType); ok {
+			body.SysType = sys.(string)
+		} else {
+			return nil, fmt.Errorf("must set %s in order to specify deployment target", Arg_SysType)
+		}
+		body.DeploymentTarget = expandDeploymentTarget(deploymentTarget.(*schema.Set).List())
+	} else {
+		if sys, ok := d.GetOk(Arg_SysType); ok {
+			body.SysType = sys.(string)
+		}
```

Comment:
You may be able to do this by using `RequiredWith` in the schema instead

Reply:
updated